backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/overlay/BpkOverlay.kt

```diff
@@ -0,0 +1,165 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.overlay
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.unit.LayoutDirection
+import net.skyscanner.backpack.compose.theme.BpkTheme
+
+enum class BpkOverlayType {
+  SolidLow,
+  SolidMedium,
+  SolidHigh,
+  TopLow,
+  TopMedium,
+  TopHigh,
+  BottomLow,
+  BottomMedium,
+  BottomHigh,
+  LeftLow,
+  LeftMedium,
+  LeftHigh,
+  RightLow,
+  RightMedium,
+  RightHigh,
+  Vignette,
+}
+
+enum class GradientLevel {
+  Low,
+  Medium,
+  High,
+}
+
+enum class GradientDirection {
+  Solid,
+  Top,
+  Bottom,
+  Left,
+  Right,
+  Vignette,
+}
+
+@Composable
+fun BpkOverlayType.toGradientlevel(): GradientLevel =
```

Comment:
No need to make it composable, plus it can be private as well?

Reply:
Fixed this! But just to understand this better, why does it not need to be Composable?

Reply:
It's just a common function, doesn't do any composable stuff inside it